contrib/endpoints/src/api_manager/api_manager_impl.cc

```diff
@@ -27,26 +27,110 @@ ApiManagerImpl::ApiManagerImpl(std::unique_ptr<ApiManagerEnvInterface> env,
     : global_context_(
           new context::GlobalContext(std::move(env), server_config)) {
   if (!service_config.empty()) {
-    AddConfig(service_config);
+    AddConfig(service_config, true);
   }
 
   check_workflow_ = std::unique_ptr<CheckWorkflow>(new CheckWorkflow);
   check_workflow_->RegisterAll();
 }
 
-void ApiManagerImpl::AddConfig(const std::string &service_config) {
+void ApiManagerImpl::AddConfig(const std::string &service_config,
+                               bool deploy_it) {
   std::unique_ptr<Config> config =
       Config::Create(global_context_->env(), service_config);
   if (config != nullptr) {
-    service_context_ = std::make_shared<context::ServiceContext>(
+    std::string config_id = config->service().id();
+    service_context_map_[config_id] = std::make_shared<context::ServiceContext>(
         global_context_, std::move(config));
+    // TODO: if this function is called at worker process, need to call
+    // service_context->service_control()->Init().
+    // ApiManagerImpl constructor is called at master process, not at worker
+    // process.
+    if (deploy_it) {
+      DeployConfigs({{config_id, 0}});
+    }
```

Comment:
LoadServiceConfig at config.cc should have logged proper info if the service config is invalid.